### ISSUE TITLE
ft_rejectcomponent error

### ISSUE THREAD

**Issue by dnacombo** [fieldtrip/fieldtrip#980]
Just after computing components, I try to reject components and get the following error.
```
cfgica              = [];
cfgica.method       = 'runica';
cfgica.numcomponent = 50;
comp             = ft_componentanalysis(cfgica, data);

cfg                           = [];
cfg.component                 = [1 10];
data_corrected                = ft_rejectcomponent(cfg, comp, data );

undoing the comp balancing for the gradiometer definition
Error using ft_datatype_sens (line 279)
unexpected number of coils (510) contributing to channel MEG0111 (1)
Error in undobalancing (line 58)
sens = ft_datatype_sens(sens);
Error in ft_senstype (line 283)
      sens = undobalancing(sens);
Error in ft_datatype_sens (line 151)
ismeg = ft_senstype(sens, 'meg');
Error in ft_datatype_raw (line 112)
      data.grad = ft_datatype_sens(data.grad);
Error in ft_datatype_comp (line 110)
      raw = ft_datatype_raw(raw, 'version', rawversion, 'hassampleinfo', hassampleinfo, 'hastrialinfo', hastrialinfo);
Error in ft_checkdata (line 266)
  data = ft_datatype_comp(data, 'hassampleinfo', hassampleinfo);
Error in ft_rejectcomponent (line 90)
  comp    = ft_checkdata(comp, 'datatype', 'comp'); 

```
Nota:
```
data.grad
ans = 
  struct with fields:

     balance: [1×1 struct]
     chanori: [306×3 double]
     chanpos: [306×3 double]
    chantype: {306×1 cell}
    chanunit: {306×1 cell}
     coilori: [510×3 double]
     coilpos: [510×3 double]
    coordsys: 'neuromag'
       label: {306×1 cell}
         tra: [306×510 double]
        type: 'neuromag306'
        unit: 'cm'

comp.grad
ans = 
  struct with fields:

       balance: [1×1 struct]
       chanori: [50×3 double]
       chanpos: [50×3 double]
       coilori: [510×3 double]
       coilpos: [510×3 double]
      coordsys: 'neuromag'
         label: {50×1 cell}
           tra: [50×510 double]
          unit: 'cm'
    chanposold: [306×3 double]
      labelold: {306×1 cell}
    chanoriold: [306×3 double]
```

**Comment by schoffelen** [comment]
can you be perhaps a bit more specific with respect to the fieldtrip version used? Also, perhaps ask this question on the discussion e-mail list, because only a few people are aware of the 'github issues' site, so if you expect an answer to this, you improve the probability of getting one, once posted there.

**Comment by dnacombo** [comment]
Sure. I tested it with the latest version, pulled yesterday just before submitting the issue. Had the impression it's a bug so submitted here, but i'll post to the list as well. Thanks!

**Comment by schoffelen** [comment]
The reason I ask for a message on the discussion list, is that other people might have run into this before, and therefore can contribute to addressing it.

Also, is this the first time you encounter this? In other words, did this use to work before, and has it stopped working?

Also, without any test data to reproduce the problem, there is no way for us to start looking into this. Even better would it be if you could try and do a bit more diagnostics yourself. From the error message it seems that fieldtrip chokes on how the sensor-array is represented (i.e. the 'grad' field).

**Comment by dnacombo** [comment]
Sorry for the lack of precision.
I tested it with 2 datasets recorded on our neuromag 306 system. Same error appears.
I just bisected the error and it appeared at this commit: d0a5046f21a63c238de40fd49aee2a287461db6a by @robertoostenveld 
Hard to debug further, I'm not familiar with how senstype is handled.
I put an example dataset here: https://owncloud.icm-institute.org/index.php/s/f9zIaGmH9hjcciT
The test code is here:
```
file = 'example_data.fif';

cfg  = [];
cfg.dataset = file;
cfg.channel = {'MEG' 'BIO*'};
data = ft_preprocessing(cfg);


cfgrsmp = [];
cfgrsmp.resamplefs  = 100;
cfgrsmp.detrend     = 'no';
data      = ft_resampledata(cfgrsmp, data);

cfgica              = [];
cfgica.method       = 'runica';
cfgica.numcomponent = 50;
comp             = ft_componentanalysis(cfgica, data);

cfg                           = [];
cfg.component                 = [1 10];
data_corrected                = ft_rejectcomponent(cfg, comp, data );
```

Hope this is clearer.

**Comment by schoffelen** [comment]
Hi Max, thanks for putting this back on our radar, we will look into it (trying to download the data as we speak).

**Comment by schoffelen** [comment]
@robertoostenveld perhaps you could think along, if you have some time. I can confirm Max's problem (see the error stack above), and it's a typical case of a mismatch between a (neuromag306) tra matrix (after componentanalysis based balancing) and the code-based expectation of only a single or double non-zero element per row in the tra. We have come across this in the past, but before I think of a patch, I am at a loss remembering the exact context in which this occurred earlier. Do you recall by any chance?

**Comment by robertoostenveld** [comment]
sorry for missing this sofar. I looked at it yesterday, have some ideas and will continue working on this in pull request #1160

**Comment by robertoostenveld** [comment]
this has been fixed in the master branch with #1160. If all is well, it will be part of tomorrows release as well.
  
@dnacombo please give it a try and please report back this is solved for you.

**Comment by dnacombo** [comment]
Yes, solved!
Thanks!
